src/Entity/Goal.php

```diff
@@ -0,0 +1,270 @@
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\CustomNormalizer;
+
+class Goal extends \ArrayObject
+{
+    use EntityValueTrait;
+
+    /**
+     * @param array $array
+     */
+    public function __construct(array $array = [])
+    {
+        parent::__construct($array);
+    }
+
+    /**
+     * Sets the 'id' parameter.
+     *
+     * @param string $id
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Goal
+     */
+    public function setId($id)
+    {
+        if (!is_string($id)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['id'] = $id;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'id' parameter.
+     *
+     * @return string The Identifier of the Goal
+     */
+    public function getId()
+    {
+        return $this->getEntityValue('id', '');
+    }
+
+    /**
+     * Sets the 'name' parameter.
+     *
+     * @param string $name
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Goal
+     */
+    public function setName($name)
+    {
+        if (!is_string($name)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['name'] = $name;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'name' parameter.
+     *
+     * @return string
+     *                The Name of the Goal
+     */
+    public function getName()
+    {
+        return $this->getEntityValue('name', '');
+    }
+
+    /**
+     * Sets the 'description' parameter.
+     *
+     * @param string $description
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Goal
+     */
+    public function setDescription($description)
+    {
+        if (!is_string($description)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['description'] = $description;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'description' parameter.
+     *
+     * @return string The Description of the Goal
+     */
+    public function getDescription()
+    {
+        return $this->getEntityValue('description', '');
+    }
+
+    /**
+     * Sets the 'rule_ids' parameter.
+     *
+     * @param array $ruleIds
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Goal
+     */
+    public function setRuleIds(array $ruleIds)
+    {
+        if (Utility::arrayDepth($ruleIds) > 1) {
+            throw new LiftSdkException('Rule Ids argument is more than 1 level deep.');
+        }
+
+        // Set only the array values to the rule_ids property.
+        $this['rule_ids'] = array_values($ruleIds);
```

Comment:
You probably already know all these but - Drupal 8 uses snake_case variable names and camelCased function names, which is following [PHP official recommendation](http://git.php.net/?p=php-src.git;a=blob_plain;f=CODING_STANDARDS;hb=HEAD). However, the vast majority PHP community has moved towards all camelCase (Symfony/Zend/Laravel and pretty much all of the ones that matters). Given the fact that this is a new, flashy, general-purpose, and dev-facing SDK, I'd personally just use camelCase for all. So I'd recommend no "rule_ids", just "ruleIds".